thingooConnector/httpconnector.py

```diff
@@ -0,0 +1,176 @@
+import json
+import logging
+from http import HTTPStatus
+
+import requests
+
+from thingooConnector import config
+from thingooConnector.config import REGISTER_ENDPOINT
+from thingooConnector.connector import Connector
+from thingooConnector.encoder import ComplexEncoder
+
+logger = logging.getLogger(__name__)
+
+
+class ClientCredentials:
+    """
+    A class used for store client credentials which are used to get access token
+    """
+
+    def __init__(self, client_id, client_secret):
+        self._client_id = client_id
+        self._client_secret = client_secret
+
+    def client_id(self):
+        return self._client_id
+
+    def client_secret(self):
+        return self._client_secret
+
+
+class HTTPException(Exception):
+    def __init__(self, error_text, status_code, text):
+        super().__init__(f'{error_text}: {status_code} {text}')
+
+
+class RetrieveTokenException(HTTPException):
+    def __init__(self, status_code, text):
+        super().__init__("Fail to retrieve access token", status_code, text)
+
+
+class RegisterDeviceException(HTTPException):
+    def __init__(self, status_code, text):
+        super().__init__("Fail to register device", status_code, text)
+
+
+class Token:
+    """
+    A class used for store token
+    """
+
+    def __init__(self, json):
+        """
+        Create token from json
+        :param json: A dict with token fields
+        """
+        self._access_token = json["access_token"]
+        self._expires_in = json["expires_in"]
+
+    def access_token(self):
+        return self._access_token
+
+
+class HTTPConnector(Connector):
+    """
+    Connector class to Thingoo instance
+    """
+
+    def __init__(self, device_info, host, client_credentials, entities):
+        super().__init__(host, device_info, entities)
+        self._client_credentials = client_credentials
+        self._token = None
+
+    def connect(self):
+        """
+        Connect to Thingoo instance: get token and register device
+        """
+        logger.info(f'Connecting to {self._host}...')
+        # Get OAuth token or raise RetrieveTokenException
+        self._token = self._get_token()
+        # Register device or raise RegisterDeviceException
+        self._register()
+        logger.info(f'Device connected!')
+
+    def _update_token(self):
+        """
+        Update token stored in self._token
+        """
+        try:
+            self._token = self._get_token()
+            logger.info("New OAuth token retrieved")
```

Comment:
I don't think you can call it an 'OAuth' token. I would print "New JWT retrieved" instead. Same for the exception below.

Reply:
Resolved in 64a07120236722cae48a65bd32e8aeaff0c88d37